tests/queries/0_stateless/03355_mergetree_table_disk.sql

```diff
@@ -0,0 +1,40 @@
+-- Tags: no-parallel, no-fasttest
+-- Tag no-parallel - uses external data source
+-- Tag no-fasttest - requires SSL for https
+
+DROP TABLE IF EXISTS uk_price_paid;
+
+-- table_disk is supported only by s3_plain/s3_plain_rewritable/web
+CREATE TABLE test_table_disk_requires_disk (key Int) ENGINE=MergeTree ORDER BY () SETTINGS table_disk=1; -- { serverError BAD_ARGUMENTS }
+CREATE TABLE test_table_disk_requires_proper_disk (key Int) ENGINE=MergeTree ORDER BY () SETTINGS disk='default', table_disk=1; -- { serverError BAD_ARGUMENTS }
+
+CREATE TABLE uk_price_paid
+(
+    price UInt32,
+    date Date,
+    postcode1 LowCardinality(String),
+    postcode2 LowCardinality(String),
+    type Enum8('other' = 0, 'terraced' = 1, 'semi-detached' = 2, 'detached' = 3, 'flat' = 4),
+    is_new UInt8,
+    duration Enum8('unknown' = 0, 'freehold' = 1, 'leasehold' = 2),
+    addr1 String,
+    addr2 String,
+    street LowCardinality(String),
+    locality LowCardinality(String),
+    town LowCardinality(String),
+    district LowCardinality(String),
+    county LowCardinality(String)
+)
+ENGINE = MergeTree
+ORDER BY (postcode1, postcode2, addr1, addr2)
+SETTINGS disk = disk(type = web, endpoint = 'https://raw.githubusercontent.com/ClickHouse/web-tables-demo/main/web/store/cf7/cf712b4f-2ca8-435c-ac23-c4393efe52f7/'), table_disk=1;
```

Comment:
Interesting if we can avoid introducing the setting and automatically detect this (for example, if it contains a proper `format_version.txt` without adding a relative path from the database - then it's `table_disk=1`)

Reply:
>Interesting if we can avoid introducing the setting and automatically detect this (for example, if it contains a proper format_version.txt without adding a relative path from the database - then it's table_disk=1)

I hate to have settings for everything, but it is reality...

As for your comment, it is a nice idea to detect it automatically, but the setting still can be useful, i.e. in case of user do not have `format_version.txt` (AFAIR it will be restored) or it has a mess in files, and by some reason `format_version.txt` is also there

Reply:
Plus, this file could be removed and the after DETACH/ATTACH the behavior will be different, so we need to save this info somewhere - settings is a good storage for this I think

Reply:
>it is a nice idea to detect it automatically

But let's do it separately, since this will require to always add `table_disk` into `SETTINGS` (so as `index_granualrity` now), which will change tons of reference files at least, and I'm not sure that it worth, maybe there will be a better way later